### PR TITLE
Implement RpcApi::get_net_totals()

### PR DESCRIPTION
**Pull request by shesek**

### CODE REVIEW

File: json/src/lib.rs [hunk 1]
```diff
@@ -1124,6 +1124,39 @@ pub struct SignRawTransactionInput {
...
+    /// Upload target statistics
+    #[serde(rename = "uploadtarget")]
+    pub upload_target: GetNetTotalsResultUploadTarget,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+pub struct GetNetTotalsResultUploadTarget {
+    /// Length of the measuring timeframe in seconds
+    #[serde(rename = "timeframe")]
+    pub time_frame: u64,
```
Comment (sgeisler) [inline json/src/lib.rs:1147]: Sounds like this should be a `Duration` except if we handle time the same way elsewhere.

File: json/src/lib.rs [hunk 2]
```diff
@@ -1124,6 +1124,39 @@ pub struct SignRawTransactionInput {
...
+    /// Target in bytes
+    pub target: u64,
+    /// True if target is reached
+    pub target_reached: bool,
+    /// True if serving historical blocks
+    pub serve_historical_blocks: bool,
+    /// Bytes left in current time cycle
+    pub bytes_left_in_cycle: u64,
+    /// Seconds left in current time cycle
+    pub time_left_in_cycle: u64,
```
Comment (sgeisler) [inline json/src/lib.rs:1157]: same `Duration` thing

**Review by sgeisler — APPROVED** [review]

File: json/src/lib.rs (hunk 1 above)
Comment (sgeisler) [reply json/src/lib.rs:1147]: Ah, I see there's a TODO comment about that but time is still handle as timestamps and seconds. Maybe we should discuss this at some point.

**Comment by shesek** [comment]
Rebased and added a (very) simple test.

**Review by stevenroose — APPROVED** [review]

**Pull request merged**